### DISCUSSION TITLE
Generate all PDF in just one page

### DISCUSSION THREAD

**Discussion by lmmartinb** [mpdf/mpdf#2001] [Q&A]
Hi, I'm trying to create a PDF file to use in a thermal printer, so I need only one page that height will increase depending on the information.

I'm completely blocked with this feature, so it's possible to do that with MPDF?

Thanks in advance.

**Comment by finwe** [comment]
I don't think this is possible with mPDF, the page size is inherently required for the created document. 

You could set page height to an obscene amount and then crop the empty bottom space from the resulting PDF by an external program

**Comment by jakejackson1** [comment]
You could probably make this work by:

1. Set mPDF to use a standard paper size (like A4) and no page margins.
2. Add your HTML to the document
3. Read the total number of pages from the mPDF object
4. Calculate the exact height required based off the paper size height * number of pages + a bit extra for margins

You can then create a new mPDF object with the calculated paper size and generate the PDF as you normally would.

**Reply by finwe** [reply]
This can be inaccurate when you take page breaks and overflowing content into consideration, but the resulting height should be always larger than the height actually needed, so the worst case scenario is you will have a bit of free space at the end.

Possible approach nonetheless.
